utils.go

```diff
@@ -23,3 +27,47 @@ func regexGroups(parsed [][]string, subexpNames ...string) map[string]string {
 	}
 	return groups
 }
+
+func encodeStructToQueryValues(s interface{}) (string, error) {
+	values := make(url.Values)
+	v := reflect.ValueOf(s)
+
+	for i := 0; i < v.Type().NumField(); i++ {
+
+		// don't include parameters with default or without values in the query to avoid corruption of the API response.
+		if isEmptyValue(v.Field(i)) {
```

Comment:
Since the `isEmptyValue(...)` function is only used here and probably won't be used in any other place, its code can be put in here directly instead of outsourcing it to a separate function